numba_dpex/tests/test_parfors.py

```diff
@@ -0,0 +1,887 @@
+from math import sqrt
+import re
+import dis
+import numbers
+import os
+import platform
+import sys
+import subprocess
+import types as pytypes
+import warnings
+from functools import reduce
+import numpy as np
+from numpy.random import randn
+import operator
+from collections import defaultdict, namedtuple
+import copy
+from itertools import cycle, chain
+import subprocess as subp
+
+import numba.parfors.parfor
+from numba import (njit, prange, set_num_threads, get_num_threads, typeof)
+from numba.core import (types, utils, typing, errors, ir, rewrites,
+                        typed_passes, inline_closurecall, config, compiler, cpu)
+from numba.extending import (overload_method, register_model,
+                             typeof_impl, unbox, NativeValue, models)
+from numba.core.registry import cpu_target
+from numba.core.annotations import type_annotations
+from numba.core.ir_utils import (find_callname, guard, build_definitions,
+                            get_definition, is_getitem, is_setitem,
+                            index_var_of_get_setitem)
+from numba.np.unsafe.ndarray import empty_inferred as unsafe_empty
+from numba.core.bytecode import ByteCodeIter
+from numba.core.compiler import (compile_isolated, Flags, CompilerBase,
+                                 DefaultPassBuilder)
+from numba.core.compiler_machinery import register_pass, AnalysisPass
+from numba.core.typed_passes import IRLegalization
+from numba.tests.support import (TestCase, captured_stdout, MemoryLeakMixin,
+                      override_env_config, linux_only, tag,
+                      skip_parfors_unsupported, _32bit, needs_blas,
+                      needs_lapack, disabled_test, skip_unless_scipy,
+                      needs_subprocess)
+from numba.core.extending import register_jitable
+import cmath
+import unittest
```

Comment:
@diptorupd this code was done using unittest, do we need to refactor it to be run with pytest?

Reply:
No, let us keep it as unittest. Pytest is able to run unittest test cases, and preferably we want to keep using the Numba source with minimal or no modification to lower maintenance overhead.